magicctapipe/scripts/lst1_magic/semi_automatic_scripts/coincident_events.py

```diff
@@ -0,0 +1,253 @@
+
+    conf = {
+        "mc_tel_ids": config_gen["mc_tel_ids"],
+        "event_coincidence": config_dict["event_coincidence"],
+    }
+
+    file_name = f"{target_dir}/v{__version__}/{source_name}/config_coincidence.yaml"
+
+    with open(file_name, "w") as f:
+
+        yaml.dump(conf, f, default_flow_style=False)
+
+
+def linking_bash_lst(target_dir, LST_runs, source_name, LST_version, env_name, cluster):
+
+    """
+    This function links the LST data paths to the working directory and creates bash scripts.
+
+    Parameters
+    ----------
+    target_dir : str
+        Path to the working directory
+    LST_runs : matrix of strings
+        This matrix ([['date','run'],['date','run']...]) is imported from *_LST_runs.txt files and tells the function where to find the LST data and link them to our working directory
+    source_name : str
+        Target name
+    LST_version : str
+        The lstchain version used to process the LST data
+    env_name : str
+        Name of the conda environment
+    cluster : str
+        Cluster system
+    """
+
+    coincidence_DL1_dir = f"{target_dir}/v{__version__}/{source_name}"
+
+    MAGIC_DL1_dir = f"{target_dir}/v{__version__}/{source_name}/DL1"
+
+    dates = [os.path.basename(x) for x in glob.glob(f"{MAGIC_DL1_dir}/Merged/[0-9]*")]
+    if cluster != "SLURM":
+        logger.warning(
+            "Automatic processing not implemented for the cluster indicated in the config file"
+        )
+        return
+    for d in dates:
+        Y_M, M_M, D_M = [int(x) for x in d.split("_")[:]]
+
+        day_MAGIC = dtdt(Y_M, M_M, D_M)
+
+        delta = timedelta(days=1)
+        for i in LST_runs:
+            Y_L, M_L, D_L = [int(x) for x in i[0].split("_")]
+
+            day_LST = dtdt(int(Y_L), int(M_L), int(D_L))
+            if day_MAGIC == day_LST + delta:
+
+                lstObsDir = i[0].replace("_", "")
+                inputdir = (
+                    f"/fefs/aswg/data/real/DL1/{lstObsDir}/{LST_version}/tailcut84"
+                )
+
+                outputdir = f"{coincidence_DL1_dir}/DL1Coincident/{lstObsDir}"
+                os.makedirs(f"{outputdir}/logs", exist_ok=True)
+
+                list_of_subruns = np.sort(glob.glob(f"{inputdir}/dl1*Run*{i[1]}*.*.h5"))
+
+                with open(f"{outputdir}/logs/list_LST.txt", "a+") as LSTdataPathFile:
+                    for subrun in list_of_subruns:
+                        LSTdataPathFile.write(f"{subrun}\n")
+
+                if not os.path.exists(f"{outputdir}/logs/list_LST.txt"):
+                    continue
+                with open(f"{outputdir}/logs/list_LST.txt", "r") as f:
+                    process_size = len(f.readlines()) - 1
+
+                if process_size < 0:
+                    continue
+                slurm = slurm_lines(
+                    queue="short",
+                    job_name=f"{source_name}_coincidence",
+                    array=process_size,
+                    mem="8g",
```

Comment:
from the tests done by @joannaWojS coincidence script was using on average 1 G of memory and the largest files were 3.4G, so I think we can lower it a bit, how about 4.5g? Or did you get any files where more memory than this was required?

Reply:
On a test I got to a maximum about 5 g, so maybe we can reduce it to 6 but I won't reduce it to 4.5 or some jobs will fail and will have to be launched again